internal/load/load.go

```diff
@@ -291,6 +291,7 @@ type API struct {
 	PercToDecimal    bool              `yaml:"perc_to_decimal"` // will check strings, and perform a trimRight for the %
 	PluckNumbers     bool              `yaml:"pluck_numbers"`   // plucks numbers out of the value
 	Math             map[string]string `yaml:"math"`            // perform match across processed metrics
+	MathDefault      string            `yaml:"math_default"`    // if unable to substitute a math variable use this default
```

Comment:
@Kav, I think it would make more sense for MathDefault to be a property of Math itself instead. Math would have to be a struct like Parse (SubParse).
Then, config could be like:

    - name: someService
      url: http://some-service.com/status
      math:
          sum: ${leaderInfo.abc.def} + ${leaderInfo.abc.hij} + 1
          default: 1

which i think makes more sense. Otherwise, the user may just put math_default anywhere and it wouldn't do anything ant it wouldn't also fail (this is something that happens in a few other cases that we should fix sometime).

Another option could be having the option of defining default values for each of the expressions, a bit like yaml itself supports doing. 

    ${leaderInfo.abc.def:<default_value>}

For example:

    - name: someService
      url: http://some-service.com/status
      math:
          sum: ${leaderInfo.abc.def:0} + ${leaderInfo.abc.hij:0} + 1

This way we wouldn't need the **default**

Reply:
@ardias The first suggestion would not support multiple variables, and would be a breaking change.

I think your second idea is the best solution. I'll rework this PR to support the second approach. Thx

Reply:
@Kav91 Since I guess the docs will change following your rework, ping me here once `functions.md` has been edited further and I'll gladly step in to review it.

Reply:
@Kav91 the first suggestion would give a default value for the whole "equation" not to the individual values. So if any part of the equation could not be evaluated (or return an error) you would return the default value. The way you have it now, the default seems disconnected from math (apart from the name)